raftwal/wal.go

```diff
@@ -104,6 +105,16 @@ func (l *wal) allEntries(lo, hi, maxSize uint64) []raftpb.Entry {
 	return entries
 }
 
+func (l *wal) truncateEntriesUntil(lastIdx uint64) error {
+	entries := l.allEntries(0, math.MaxUint64, math.MaxUint64)
```

Comment:
This function copies data rather than referencing it from the mmap file, so this operation would not be light on memory use.